src/gpu/intel/jit/config/ngen_config.hpp

```diff
@@ -41,7 +41,8 @@ using half = dnnl::impl::float16_t;
 #define NGEN_HALF_TYPE
 
 #if (!defined(NDEBUG) || defined(DNNL_DEV_MODE)) \
-        && (__cplusplus >= 202002L || _MSVC_LANG >= 202002L)
+        && ((defined(__cplusplus) && __cplusplus >= 202002L) \
```

Comment:
Do we really need to check if `__cplusplus` is defined? According to https://en.cppreference.com/w/cpp/preprocessor/replace#Predefined_macros, it is a predefined macro and so it will always be defined.

Reply:
I think I can drop it unless there's a case when this header is going to be used with plain C compiler, in which case this macro is not defined.

Reply:
Removed.